### DISCUSSION TITLE
performance issues

### DISCUSSION THREAD

**Discussion by jh-ph** [utmstack/UTMStack#575] [Q&A] [answered]
Hi,
I'm running UTMStack 10.3.0 in a Virtualbox VM. Manual installation of UTMStack in Ubuntu. The VM previously had 16GB of RAM, we increased it to 32GB. Still see sluggish performance when clicking around in the dashboard. Below are some screenshots of the resource usage. Integrations include, O365, one Windows Agent for now on a tertiary Windows Domain Controller, Fortinet/Fortigate, Sophos (not working yet will need to troubleshoot), and a SOC AI integration. Tried sending our Fortimanager syslog data to the UTMStack, and it slowed down the dashboard even more, maximizing the RAM usage.

Just wondering why the RAM only seems to show only 6GB available to UTMStack, can we increase that? The VM has 32GB of RAM and the VM usage shows about 16GB. So there is available RAM to be assigned to the application.

![blank_Screenshot 2024-03-19 120957](https://github.com/utmstack/UTMStack/assets/160979607/7189f8ab-58cf-4942-b1df-a9e8069a2e45)
![dataeng_2024-03-19 121025](https://github.com/utmstack/UTMStack/assets/160979607/0d87f72b-572a-416d-9593-eb874b145fd0)
![dataproc_2024-03-19 120906](https://github.com/utmstack/UTMStack/assets/160979607/78d0cb84-a5b2-4ca4-99ea-b40189cc50b2)

HTOP:
![image](https://github.com/utmstack/UTMStack/assets/160979607/0e389970-9157-4e48-952d-cf7a5c51a990)

Thanks,
JH

**Comment by c3s4rfred** [comment, marked as answer]
Hi, @jh-ph we will check about that, but you can try to reboot the machine and then if still don't see the memory increased on the view, run the installer again, in order to UTMStack to be aware of the increased memory.

**Comment by jh-ph** [comment]
Hi @c3s4rfred , thank you for the assistance. Had to run the installer again for UTMStack to see the additional memory allocated to the VM. But still experiencing performance issues. Sometimes clicking on links will just continue to load for extended period of time, from 10 seconds to 30 seconds. Sometimes I would need to click around to other links and back to get to the page I wanted to go to. Other times, I would need ot manually refresh the page. Just wanted to know if that's normal behavior. Or do we need to add more memory to the vm? VM usage is now about 10-12GB out of 14GB. 
When this happens, the links that shows the resource usage also displays blank windows:
![image](https://github.com/utmstack/UTMStack/assets/160979607/c4bdabfd-24d9-4912-985e-95ae1491f91a)

Clicking other links won't do anything, need to wait about 10-20 seconds and then we can start to click around again. This happens a lot.

Thanks
JH

**Reply by c3s4rfred** [reply]
Hi, @jh-ph there is a new version released, please update your UTMStack instance. Normally, the resource consumption is related to the amount of integrations and incoming logs per second to UTMStack, tell us that number if you have it at hand.
